core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2132,6 +2136,93 @@ private[spark] object Utils extends Logging {
       .getOrElse(UserGroupInformation.getCurrentUser().getShortUserName())
   }
 
+  /**
+   * Adds a shutdown hook with default priority.
+   */
+  def addShutdownHook(hook: () => Unit): AnyRef = {
```

Comment:
It looked weird to return `AnyRef`, but I suppose that while callers need a handle to the hook, they don't need to be promised anything about what it is.